techmatters/hrm
# [PoC] Import transcripts of text conversations

## Description
This PR is a PoC for https://benetech.app.box.com/file/988368911133.
Processor Lambda code in https://github.com/techmatters/infrastructure-config/pull/67

This PR is kinda branched off of https://github.com/techmatters/hrm/pull/209, replacing:
- Type of job is decided using the `channel` column of a contact (is `default` for offline contacts).
- Local processing of due jobs by sending them to an SNS topic.
- El cheapo simulation of a 'completed jobs' SQS queue by the actual SQS queue message.

The transcripts urls are saved in the same naively way as in the original PoC using `appendMediaUrls`. 
Should we migrate the table to add an extra column?
Does it makes sense to migrate `channelSid` to something more universal like `conversationSid` to use in all type of contacts (except for offline contacts of course)?

### Checklist

### Verification steps
- Add to your `.env` the following parameters:
  - `AWS_ACCESS_KEY_ID`=your user key
   - `AWS_SECRET_ACCESS_KEY`=your secret
   - `AWS_REGION`=the default region (where dev is hosted)
   - `JOBS_TOPIC_ARN`=development-jobs-topic SNS topic ARN
   - `COMPLETED_JOBS_QUEUE_URL`=development-completed-jobs SQS queue URL
- Save a few chat contacts in your database
- Check the console outputs, play around with the DB, etc. What to expect? You should see pending jobs being added alongside contacts, the job processor looping, sending SNS messages for the pending jobs and then consuming the completed jobs payloads from SQS once the Lambda has processed them. Transcript files should be created in the development docs bucket, and the url of each attached to the contact (in rawJson for now).
